pints/_mcmc/_adaptive_covariance.py

```diff
@@ -154,67 +136,52 @@ def tell(self, fx):
             # Clear proposal
             self._proposed = None
 
+            # Set r to zero
+            self._r = float('-Inf')
+
             # Return first point for chain
             return self._current
 
         # Check if the proposed point can be accepted
-        accepted = 0
-        if np.isfinite(fx):
-            u = np.log(np.random.uniform(0, 1))
-            if u < fx - self._current_log_pdf:
-                accepted = 1
-                self._current = self._proposed
-                self._current_log_pdf = fx
-
-        # Clear proposal
-        self._proposed = None
-
-        # Adapt covariance matrix
-        if self._adaptive:
-            # Set gamma based on number of adaptive iterations
-            gamma = self._adaptations ** -0.6
-            self._adaptations += 1
-
-            # Update mu, log acceptance rate, and covariance matrix
-            self._mu = (1 - gamma) * self._mu + gamma * self._current
-            self._loga += gamma * (accepted - self._target_acceptance)
-            dsigm = np.reshape(
-                self._current - self._mu, (self._n_parameters, 1))
-            self._sigma = (
-                (1 - gamma) * self._sigma + gamma * np.dot(dsigm, dsigm.T))
-
-        # Update acceptance rate (only used for output!)
-        self._acceptance = ((self._iterations * self._acceptance + accepted) /
-                            (self._iterations + 1))
+        self._accepted = 0
+        self._r = fx - self._current_log_pdf
 
-        # Increase iteration count
-        self._iterations += 1
+    def _update_mu(self):
+        """
+        Updates the current running mean used to calculate the sample
+        covariance matrix of proposals. Note that this default is overidden in
+        some of the methods
+        """
+        raise NotImplementedError
 
-        # Return new point for chain
-        return self._current
+    def _update_sigma(self):
+        """
+        Updates the covariance matrix used to generate proposals.
+        Note that this default is overidden in some of the methods
+        """
+        raise NotImplementedError
 
-    def replace(self, current, current_log_pdf, proposed=None):
+    def replace(self, x, fx):
```

Comment:
I think we needed to be able to update the proposal to make SMC work.
(If we decide to drop SMC I suppose this is fine?)

Reply:
Either way, the replace method implementation can wait until after this is done, in my opinion :-|

Reply:
I've left it in for now. Can discuss later...